src/mantine-dates/src/components/RangeCalendar/RangeCalendar.tsx

```diff
@@ -120,68 +125,135 @@ export const RangeCalendar = forwardRef<HTMLDivElement, RangeCalendarProps>(
+      hasMultipleMonths: withMultipleMonths,
+    });
+
+    const dayStyles = (date, modifiers, isSecondMonth = false) => {
+      const initialStyles = typeof dayStyle === 'function' ? dayStyle(date, modifiers) : {};
+      if (
+        shouldHighlightDate(
+          date,
+          modifiers,
+          dayjs(_month)
+            .add(isSecondMonth ? 1 : 0, 'month')
+            .toDate()
+        )
+      ) {
+        return {
+          ...initialStyles,
+          backgroundColor:
+            theme.colorScheme === 'dark'
+              ? theme.fn.rgba(theme.colors[theme.primaryColor][9], 0.3)
+              : theme.colors[theme.primaryColor][0],
+          borderRadius: 0,
+        };
+      }
+
+      return initialStyles;
+    };
 
     return (
       <CalendarWrapper
         size={size}
         fullWidth={fullWidth}
+        hasMultipleMonths={withMultipleMonths}
         onMouseLeave={handleMouseLeave}
         ref={ref}
         {...others}
       >
-        <CalendarHeader
-          size={size}
-          nextMonthLabel={nextMonthLabel}
-          previousMonthLabel={previousMonthLabel}
-          previousMonthDisabled={disabledState.previousDisabled}
-          nextMonthDisabled={disabledState.nextDisabled}
-          onPreviousMonth={() => setMonth(dayjs(_month).subtract(1, 'month').toDate())}
-          onNextMonth={() => setMonth(dayjs(_month).add(1, 'month').toDate())}
-          classNames={classNames}
-          styles={styles}
-          locale={locale}
-          withSelect={withSelect}
-          yearsRange={yearsRange}
-          month={_month}
-          setMonth={setMonth}
-          labelFormat={labelFormat}
-          __staticSelector={__staticSelector}
-        />
-
-        <Month
-          month={_month}
-          range={value}
-          value={pickedDate}
-          onChange={setRangeDate}
-          dayClassName={dayClassName}
-          dayStyle={(date, modifiers) => {
-            const initialStyles = typeof dayStyle === 'function' ? dayStyle(date, modifiers) : {};
-            if (shouldHighlightDate(date, modifiers)) {
-              return {
-                ...initialStyles,
-                backgroundColor:
-                  theme.colorScheme === 'dark'
-                    ? theme.fn.rgba(theme.colors[theme.primaryColor][9], 0.3)
-                    : theme.colors[theme.primaryColor][0],
-                borderRadius: 0,
-              };
-            }
-
-            return initialStyles;
-          }}
-          disableOutsideEvents={disableOutsideEvents}
-          minDate={minDate}
-          maxDate={maxDate}
-          excludeDate={excludeDate}
-          classNames={classNames}
-          styles={styles}
-          fullWidth={fullWidth}
-          size={size}
-          onDayMouseEnter={(date) => setHoveredDay(date)}
-          firstDayOfWeek={firstDayOfWeek}
-          __staticSelector={__staticSelector}
-        />
+        <Group noWrap style={{ alignItems: 'flex-start' }}>
```

Comment:
there is a lot of shared props across these 2 pickers. Will making a `sharedMonthProps` and `sharedHeaderProps`  and spreading them would make it a bit more clearer? By looking that these 2 it's hard to find exact differences. @rtivital can you refer to this idea? Rendering as an array is also a point but I think it will add another complexity